auditors/mounts/mounts.go

```diff
@@ -0,0 +1,113 @@
+
+const (
+	// SensitivePathsMounted occurs when a container has sensitive host paths mounted
+	SensitivePathsMounted = "SensitivePathsMounted"
+)
+
+// DefaultSensitivePaths is the default list of sensitive mount paths (from Falco rule: https://github.com/falcosecurity/falco/blob/master/rules/k8s_audit_rules.yaml#L155)
+var DefaultSensitivePaths = []string{"/proc", "/var/run/docker.sock", "/", "/etc", "/root", "/var/run/crio/crio.sock", "/home/admin", "/var/lib/kubelet", "/var/lib/kubelet/pki", "/etc/kubernetes", "/etc/kubernetes/manifests"}
+
+const overrideLabelPrefix = "allow-host-path-mount-"
+
+// SensitivePathMounts implements Auditable
+type SensitivePathMounts struct {
+	sensitivePaths map[string]bool
+}
+
+func New(config Config) *SensitivePathMounts {
+	paths := make(map[string]bool)
+	for _, path := range config.GetSensitivePaths() {
+		paths[path] = true
+	}
+	return &SensitivePathMounts{
+		sensitivePaths: paths,
+	}
+}
+
+// Audit checks that the container does not have the Docker socket mounted
+func (sensitive *SensitivePathMounts) Audit(resource k8stypes.Resource, _ []k8stypes.Resource) ([]*kubeaudit.AuditResult, error) {
+	var auditResults []*kubeaudit.AuditResult
+
+	spec := k8s.GetPodSpec(resource)
+	if spec == nil {
+		return auditResults, nil
+	}
+
+	sensitiveVolumes := auditPodVolumes(spec, sensitive.sensitivePaths)
+
+	if sensitiveVolumes == nil || len(sensitiveVolumes) == 0 {
+		return auditResults, nil
+	}
+
+	for _, container := range k8s.GetContainers(resource) {
+		for _, auditResult := range auditContainer(container, sensitiveVolumes) {
+			auditResult = override.ApplyOverride(auditResult, container.Name, resource, getOverrideLabel(auditResult.Metadata["Mount"]))
+			if auditResult != nil {
+				auditResults = append(auditResults, auditResult)
+			}
+		}
+	}
+
+	return auditResults, nil
+}
+
+func auditPodVolumes(podSpec *k8stypes.PodSpecV1, sensitivePaths map[string]bool) map[string]v1.Volume {
+	if podSpec.Volumes == nil {
+		return nil
+	}
+
+	found := make(map[string]v1.Volume, 0)
+	for _, volume := range podSpec.Volumes {
+		if volume.HostPath == nil {
+			continue
+		}
+
+		if _, ok := sensitivePaths[volume.HostPath.Path]; ok {
+			found[volume.Name] = volume
+		}
+	}
+
+	return found
+}
+
+func auditContainer(container *k8stypes.ContainerV1, sensitiveVolumes map[string]v1.Volume) []*kubeaudit.AuditResult {
+	if container.VolumeMounts == nil {
+		return nil
+	}
+
+	var auditResults []*kubeaudit.AuditResult
+
+	for _, mount := range container.VolumeMounts {
+		if _, ok := sensitiveVolumes[mount.Name]; ok {
+			volume := sensitiveVolumes[mount.Name]
+				auditResults = append(auditResults, &kubeaudit.AuditResult{
+					Name:     SensitivePathsMounted,
+					Severity: kubeaudit.Error,
+					Message:  fmt.Sprintf("Sensitive path mounted as volume: %s (%s -> %s, readOnly: %t). It should be removed from the container's mounts list.", mount.Name, volume.HostPath.Path, mount.MountPath, mount.ReadOnly),
```

Comment:
I find both to be valid :) My reasoning in putting the host path first was that this auditor is looking for sensitive host paths and the message is "Sensitive path mounted as volume" so I though it would make sense to display the sensitive path first. But I'm totally ok to swap if you think it's indeed better :)

Reply:
The arrow makes me think it's directional, and having the mount path point to the host path makes more sense to me, but if even we aren't sure then this is probably going to cause confusion and I would rather just omit the mount path (it should probably be part of the result metadata anyway). The host path is the important part and the mount name should be enough to identify where the problem is

Reply:
Ok, will do that 👌  What abut the `readOnly` info then? If the `mountPath` is no more in the main message, I'm not sure it makes sense to have it there too. Should we put it in the metadata too or just remove it altogether?

Reply:
Yea good point, probably remove the `readOnly` from the message as well. I don't think it hurts to include in the metadata if you think it's useful information to know when reviewing results

Reply:
My idea for displaying this value was that you may want to put extra care in checking container that can write on the host because that a big step further in terms of privileges. As you said it won't hurt to have it in the Metadata 🤷